gen/gen.go

```diff
@@ -0,0 +1,218 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"reflect"
+	"strconv"
+	"strings"
+	"text/template"
+)
+
+type Data struct {
+	Name string
+	*Yml
+}
+
+func GenCHeader(data *Data, dst io.Writer) error {
+	t := template.
+		New("").
+		Funcs(template.FuncMap{
+			"SComment":     func(v string, indent int) string { return Comment(v, CommentTypeSingleLine, indent, true) },
+			"MComment":     func(v string, indent int) string { return Comment(v, CommentTypeMultiLine, indent, true) },
+			"SCommentN":    func(v string, indent int) string { return Comment(v, CommentTypeSingleLine, indent, false) },
+			"MCommentN":    func(v string, indent int) string { return Comment(v, CommentTypeMultiLine, indent, false) },
+			"ConstantCase": ConstantCase,
+			"PascalCase":   PascalCase,
+			"CamelCase":    CamelCase,
+			"CType":        CType,
+			"CValue":       CValue,
+			"BitFlagValue": func(index int) uint64 { return uint64(math.Pow(2, float64(index-1))) },
+			"ParseUint": func(s string, bitSize int) uint64 {
```

Comment:
FYI: It doesn't actually look like this is ever called.

Reply:
`ParseUint` is used for enum & bitflags blocks.

Reply:
I see that it's used in the templates, but it doesn't ever appear to be called. You can replace the body with a `panic("!")` and see that it is never called.

Reply:
Correct, it'll only get called if in the yaml an enum/bitflag entry has a custom `value` property, which is optional.
Current `webgpu.yml` doesn't need them currently, but other custom specs may need them.

Reply:
👍🏼 Just checking that this was intentional. SGTM